clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -1228,7 +1228,10 @@ void CodeGenFunction::EmitBoundsCheckImpl(const Expr *E, llvm::Value *Bound,
   SanitizerScope SanScope(this);
 
   llvm::DILocation *CheckDI = Builder.getCurrentDebugLocation();
-  if (ClArrayBoundsPseudoFn && CheckDI) {
+  if ((ClArrayBoundsPseudoFn ||
```

Comment:
Add TODO to clean up mllvm flag

Reply:
Done

Reply:
Could be nice to make clear that SO_ArrayBounds here is the same as on line 1252
E.g. `auto CurrentCheck = SO_ArrayBounds;`

Reply:
Added CurrentCheckKind (there is already a `Check` variable that contains the instructions)

Reply:
I would drop the Current then for symmety

Reply:
Done